pkg/util/httputil/client.go

```diff
@@ -78,6 +98,37 @@ func (c *Client) Get(ctx context.Context, url string) (resp *http.Response, err
 	return c.Do(req)
 }
 
+// Put does like http.Client.Put but uses the provided context and obeys its cancellation.
+func (c *Client) Put(
+	ctx context.Context, url string, h *RequestHeaders, body io.Reader,
+) (resp *http.Response, err error) {
+	req, err := http.NewRequestWithContext(ctx, "PUT", url, body)
+	if err != nil {
+		return nil, err
+	}
+	if h.ContentType != "" {
```

Comment:
We should either allow for `nil` headers or state the precondition in the comment.